### PR TITLE
Remove vdrtools FFI layer

### PR DESCRIPTION
**Pull request by left-arm**
Signed-off-by: Artem Mironov <artem.mironov@absa.africa>

### CODE REVIEW

**Comment by Patrik-Stas** [comment]
Very cool ❤️ Review in progress 🚧  FYI there's one merge conflict after after we got some Miroslav's work merged on `main`

 In the meantime, for better transparency, please:
- open a PR on your https://gitlab.com/left-arm/vdr-tools.git fork from `static-link` branch against `main` to enable easily checking diff and include link to that PR in this PR's description
- write up some PR description what you have changed (just high level, if there was perhaps included any other changes than swapping FFI)

Thank you!

File: libvcx/src/api_lib/api_c/wallet.rs [hunk 1]
```diff
@@ -1345,7 +1345,7 @@ pub mod tests {
         close_main_wallet().await.unwrap();
     }
 
-    #[tokio::test]
+    #[tokio::test(flavor = "multi_thread")]
```
Comment (Patrik-Stas) [inline libvcx/src/api_lib/api_c/wallet.rs:1348]: Just curious, why is this?

File: messages/src/utils/error_codes.rs [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
-            404 => MessagesError::from_msg(MesssagesErrorKind::DuplicationMasterSecret, error.message),
-            407 => MessagesError::from_msg(MesssagesErrorKind::CredDefAlreadyCreated, error.message),
-            600 => MessagesError::from_msg(MesssagesErrorKind::DuplicationDid, error.message),
-            702 => MessagesError::from_msg(MesssagesErrorKind::InsufficientTokenAmount, error.message),
-            error_code => MessagesError::from_msg(MesssagesErrorKind::LibndyError(error_code), error.message),
-        }
+    fn from(_error: IndyError) -> Self {
+        MesssagesErrorKind::InvalidState.into()
+
+        // match error.error_code as u32 {
```
Comment (Patrik-Stas) [inline messages/src/utils/error_codes.rs:14]: can be removed?

File: messages/src/utils/error_codes.rs [hunk 3]
```diff
@@ -67,44 +69,44 @@ pub fn map_indy_error_code<C: PrimInt>(error_code: C) -> u32 {
...
-        };
-
-        assert_eq!(MessagesError::from(err100).kind(), MesssagesErrorKind::InvalidLibindyParam);
-        assert_eq!(MessagesError::from(err107).kind(), MesssagesErrorKind::InvalidLibindyParam);
-        assert_eq!(MessagesError::from(err111).kind(), MesssagesErrorKind::InvalidLibindyParam);
-        // Test that RC 112 falls out of the range 100...112
-        assert_ne!(MessagesError::from(err112).kind(), MesssagesErrorKind::InvalidLibindyParam);
-    }
-}
+// #[cfg(test)]
```
Comment (Patrik-Stas) [inline messages/src/utils/error_codes.rs:72]: I think this test was copy paster from `aries-vcx` in the first place, lets remove this

Comment (Patrik-Stas) [reply messages/src/utils/error_codes.rs:72]: And in general, we should probably extract out in future some `aries-vcx-error` crate and reuse it in `aries-vcx`, `messages`, `agency_client`

**Review by Patrik-Stas — CHANGES_REQUESTED** [review]

File: aries_vcx/src/indy/utils/mod.rs [hunk 4]
```diff
@@ -40,58 +39,54 @@ impl LibindyMock {
...
-        pub wallet_handle: vdrtools::WalletHandle,
-    }
+    const TRUSTEE_SEED: &'static str = "000000000000000000000000Trustee1";
+    const WALLET_KEY: &str = "8dvfYSt5d1taSd6yJdpjq4emkwsPDDLYxkNFysFD2cZY";
+    const WALLET_KEY_DERIVATION: &str = "RAW";
 
-    pub async fn setup_wallet() -> WalletSetup {
-        let name: String = crate::utils::random::generate_random_name();
-        let wallet_config = json!({ "id": name }).to_string();
+    pub async fn run_wallet<F>(f: impl FnOnce(vdrtools::WalletHandle) -> F)
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply aries_vcx/src/indy/utils/mod.rs:49]: Or perhaps just `run` - I've noticed you called that function `run` in different mock setup

File: aries_vcx/src/indy/utils/error_codes.rs [hunk 5]
```diff
@@ -1,33 +1,33 @@
...
-            306 => VcxError::from_msg(VcxErrorKind::CreatePoolConfig, error.message),
-            404 => VcxError::from_msg(VcxErrorKind::DuplicationMasterSecret, error.message),
-            407 => VcxError::from_msg(VcxErrorKind::CredDefAlreadyCreated, error.message),
-            600 => VcxError::from_msg(VcxErrorKind::DuplicationDid, error.message),
-            702 => VcxError::from_msg(VcxErrorKind::InsufficientTokenAmount, error.message),
-            error_code => VcxError::from_msg(VcxErrorKind::LibndyError(error_code), error.message),
-        }
-    }
-}
+// impl From<IndyError> for VcxError {
```
[the remark this answers is not in the document]
Comment (left-arm) [reply aries_vcx/src/indy/utils/error_codes.rs]: yes

File: aries_vcx/src/indy/utils/mod.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (left-arm) [reply aries_vcx/src/indy/utils/mod.rs]: renamed to with_wallet

File: libvcx/src/api_lib/api_c/wallet.rs (hunk 1 above)
Comment (left-arm) [reply libvcx/src/api_lib/api_c/wallet.rs]: This particular test requires a multi-thread executor. The last `cb.receive(..).unwrap()` fails with timeout on a single thread executor.

**Review by Patrik-Stas — DISMISSED** [review]

**Review by Patrik-Stas — APPROVED** [review]

**Review by mirgee — APPROVED** [review]

**Pull request merged**